examples/stream/port_duplex.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const { Duplex } = require('stream');
+
+class PortDuplex extends Duplex {
+  #port = undefined;
+  #transfer = false;
+
+  constructor (port, options) {
+    const {
+      readable = true,
+      writable = true,
+      transfer = false
+    } = { ...options };
+    if (typeof readable !== 'boolean') {
+      throw new TypeError('readable must be a boolean');
+    }
+    if (typeof writable !== 'boolean') {
+      throw new TypeError('writable must be a boolean');
+    }
+    if (typeof transfer !== 'boolean') {
+      throw new TypeError('transfer must be a boolean');
+    }
+    super({ autoDestroy: true, readable, writable });
+    this.#port = port;
+    this.#transfer = transfer;
+    this.#port.onmessage = PortDuplex.#onmessage.bind(this);
+  }
+
+  _write (chunk, encoding, callback) {
+    if (typeof chunk === 'string') {
+      chunk = Buffer.from(chunk, encoding);
+    }
+    const transferList = this.#transfer ? [chunk.buffer] : undefined;
```

Comment:
If the buffer was allocated with allocUnsafe this will cause problems due to pooling no?

Reply:
Well, there are many ways this is problematic. I'll be updating this such that we always create a new `Uint8Array` copy of the `Buffer` then transferring that new copy. That causes the fewest headaches overall.